core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -842,6 +847,75 @@ class ControllerApis(val requestChannel: RequestChannel,
       }
   }
 
+  def handleCreateDelegationTokenRequest(request: RequestChannel.Request): CompletableFuture[Unit] = {
```

Comment:
> Is it allowed for users to send direct requests like this to the controller (bypassing the broker)?

RIght now no, but in the future maybe yes. 
One strong motivation of the KRaft design is that you can make the controllers more secure if there isn't direct communication to the controller form an external source.

Reply:
In the Zk case we will always have to do the checks in the broker (`KafkaApis.scala` or `DelegationTokenManagerZk.scala`)

In the KRaft case where the client talks directly to the kcontroller we will have to do the checks in the controller (`ControllerApis.scala` or `DelegationTokenControlManager.java`) 

In the KRaft case where the client talks to the broker and the broker forwards the request to the kcontroller we have to do the check on the connection authentication in the broker because the information about how the client to broker connection is authenticated is not forwarded. We cannot allow clients authenticated with delegation tokens to create, renew or expire delegation tokens.